.github/workflows/go.yml

```diff
@@ -0,0 +1,75 @@
+name: Go
+
+on:
+  push:
+    branches: ['*']
+    tags: ['v*']
+  pull_request:
+    branches: ['*']
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Setup Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.14.x
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Load cached dependencies
+      uses: actions/cache@v1
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+
+    - name: Install
+      run: make install
+
+    - name: Test
+      run: make test_ci
+
+    - name: Upload coverage to codecov.io
+      uses: codecov/codecov-action@v1
+
+  release:
+    runs-on: ubuntu-latest
+
+    # Only release pushes of tags starting with v and if the build succeeded.
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
+    needs: [build]
+
+    steps:
+
+    - name: Setup Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.14.x
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Load cached dependencies
+      uses: actions/cache@v1
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
```

Comment:
is there a way to reference the same steps + params as build rather than copy+paste?

Reply:
Not that i know of. I mean, if we wanted to use YAML pointers, maybe? I wondered if it was possible to share the setup between the two but there was no obvious way.

There's some discussion in https://github.community/t5/GitHub-Actions/reusing-sharing-inheriting-steps-between-jobs-declarations/td-p/37849 but I don't see resolutions incoming.